### PR TITLE
Notify on feature flag changes

### PR DESCRIPTION
**Pull request by emord**
@esoergel @orangejenny we talked about this last week. This notifies the team on any deprecated flag changes and you can also specify yourself on any you want to track yourself.

Also considered a thank you email for removed entries. maybe some positive reinforcement would be good?

### CODE REVIEW

File: corehq/apps/toggle_ui/views.py [hunk 1]
```diff
@@ -194,6 +197,26 @@ def post(self, request, *args, **kwargs):
...
+            if self.static_toggle.notification_emails:
+                emails = [
+                    "{}@{}.com".format(email, "dimagi")
+                    for email in self.static_toggle.notification_emails
+                ]
+                _assert = soft_assert(to=emails, send_to_ops=is_deprecated_toggle)
+            else:
+                _assert = soft_assert(send_to_ops=is_deprecated_toggle)
+
+            _assert(False, subject)
```
Comment (esoergel) [inline corehq/apps/toggle_ui/views.py]: You could stick this stuff in a private method to keep the `post` method relatively small.

**Review by esoergel — APPROVED** [review]

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/toggle_ui/views.py [hunk 2]
```diff
@@ -194,6 +197,26 @@ def post(self, request, *args, **kwargs):
         toggle.save()
 
         changed_entries = previously_enabled ^ currently_enabled  # ^ means XOR
+
+        added_entries = currently_enabled - previously_enabled
+        is_deprecated_toggle = (self.static_toggle.tag == TAG_DEPRECATED)
```
Comment (orangejenny) [inline corehq/apps/toggle_ui/views.py]: I could imagine this also applying to TAG_CUSTOM and TAG_INTERNAL - anecdotally, I feel like a number of the issues we have are when a flag built for a project like ICDS starts getting used on solutions projects. Or would that lead to too many emails?

**Comment by orangejenny** [comment]
I like the ability to subscribe to individual flags. You're going to announce this to the dev team, yes?

**Comment by orangejenny** [comment]
> Also considered a thank you email for removed entries.

I am mildly curious if anyone ever turns *off* a  feature flag.

**Review by czue — APPROVED** [review]

File: corehq/apps/toggle_ui/views.py (hunk 1 above)
Comment (emord) [reply corehq/apps/toggle_ui/views.py]: https://github.com/dimagi/commcare-hq/pull/21273/commits/4a56ebeba9e34b390f4a88e6c71641fa6aec98c7

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/toggle_ui/views.py [hunk 3]
```diff
@@ -192,7 +192,7 @@ def post(self, request, *args, **kwargs):
                 _clear_caches_for_dynamic_toggle(self.toggle_meta())
 
         elif save_randomness:
-            messages.error(request, _("The randomness value {} must be between 0 and 1".format(randomness)))
```
Comment (esoergel) [inline corehq/apps/toggle_ui/views.py]: :+1:

**Review by esoergel — APPROVED** [review]

**Comment by emord** [comment]
@dimagi/team-commcare-hq You can now subscribe to feature flag additions. 

This PR allows you to add your username to a feature flag and it will email you whenever someone adds a domain or user to it. This way you can proactively track any usage for a feature flag that is not intended for general availability.

In addition if anyone adds someone to an internal, custom or deprecated feature flag, it will send an email to the team.

**Pull request merged**